pythonforandroid/recipes/python3/__init__.py

```diff
@@ -0,0 +1,198 @@
+from pythonforandroid.recipe import TargetPythonRecipe, Recipe
+from pythonforandroid.toolchain import shprint, current_directory
+from pythonforandroid.patching import (is_darwin, is_api_gt,
+                                       check_all, is_api_lt, is_ndk)
+from pythonforandroid.logger import logger, info
+from pythonforandroid.util import ensure_dir, walk_valid_filens
+from os.path import exists, join, realpath, dirname
+from os import environ, listdir, walk
+import glob
+import sh
+
+
+STDLIB_DIR_BLACKLIST = {
+    '__pycache__',
+    'test',
+    'tests',
+    'lib2to3',
+    'ensurepip',
+    'idlelib',
+    'tkinter',
+}
+
+STDLIB_FILEN_BLACKLIST = [
+    '*.pyc',
+    '*.exe',
+    '*.whl',
+]
+
+# TODO: Move to a generic location so all recipes use the same blacklist
+SITE_PACKAGES_DIR_BLACKLIST = {
+    '__pycache__',
+    'tests'
+}
+
+SITE_PACKAGES_FILEN_BLACKLIST = []
+
+
+class Python3Recipe(TargetPythonRecipe):
+    version = '3.7.1'
+    url = 'https://www.python.org/ftp/python/{version}/Python-{version}.tgz'
+    name = 'python3'
+
+    depends = ['hostpython3']
+    conflicts = ['python3crystax', 'python2']
+    # opt_depends = ['openssl', 'sqlite3']
+
+    def build_arch(self, arch):
+        recipe_build_dir = self.get_build_dir(arch.arch)
+        
+        # Create a subdirectory to actually perform the build
+        build_dir = join(recipe_build_dir, 'android-build')
+        ensure_dir(build_dir)
+
+        # TODO: Get these dynamically, like bpo-30386 does
+        sys_prefix = '/usr/local'
+        sys_exec_prefix = '/usr/local'
+
+        # Skipping "Ensure that nl_langinfo is broken" from the original bpo-30386
+
+        platform_name = 'android-{}'.format(self.ctx.ndk_api)
+
+        with current_directory(build_dir):
+            env = environ.copy()
+
+            # TODO: Get this information from p4a's arch system
```

Comment:
indeed, most* of this should be available in `arch.get_env()` right?

Reply:
Yes, but it needs some modification to work seamlessly with clang, and that effort should also have an eye to making clang work for the other recipes that still use gcc. It shouldn't be hard and I originally intended to do it in this PR, but since the redundant code is confined to the new python3 recipe I don't think it's worth holding things up for it. I'd feel differently and want to do it first if it weren't for the API 26+ problem.